scripts/st2bootstrap-deb.sh

```diff
@@ -339,17 +339,17 @@ configure_st2_user () {
     sudo useradd stanley
   fi
 
-  sudo mkdir -p /home/stanley/.ssh
+  sudo mkdir -p ~stanley/.ssh
```

Comment:
OK, did some quick tests, and this is indeed a problem. 

First I changed `/etc/st2/st2.conf` to include this:
```
[system_user]
user = stanley
ssh_key_file = ~stanley/.ssh/stanley_rsa
```
I then ran `st2 run core.remote hosts=127.0.0.1 cmd='id'`, and got this output:
```
id: 58b834b5c4da5f3126df0aaa
status: failed
parameters:
  cmd: id
  hosts: 127.0.0.1
result:
  error: "Unable to connect to any one of the hosts: [u'127.0.0.1'].

 connect_errors={
  "127.0.0.1": {
    "failed": true,
    "traceback": "Traceback (most recent call last):\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/parallel_ssh.py\", line 243, in _connect\n    client.connect()\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/paramiko_ssh.py\", line 138, in connect\n    self.client = self._connect(host=self.hostname, socket=self.bastion_socket)\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/paramiko_ssh.py\", line 592, in _connect\n    passphrase_reqd = self._is_key_file_needs_passphrase(self.key_files)\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/paramiko_ssh.py\", line 672, in _is_key_file_needs_passphrase\n    cls.from_private_key_file(file, password=None)\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/paramiko/pkey.py\", line 196, in from_private_key_file\n    key = cls(filename=filename, password=password)\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/paramiko/rsakey.py\", line 46, in __init__\n    self._from_private_key_file(filename, password)\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/paramiko/rsakey.py\", line 173, in _from_private_key_file\n    data = self._read_private_key_file('RSA', filename, password)\n  File \"/opt/stackstorm/st2/lib/python2.7/site-packages/paramiko/pkey.py\", line 267, in _read_private_key_file\n    with open(filename, 'r') as f:\nIOError: [Errno 2] No such file or directory: '~stanley/.ssh/stanley_rsa'\n",
    "timeout": false,
    "succeeded": false,
    "stdout": "",
    "stderr": "",
    "error": "Failed connecting to host 127.0.0.1. ",
    "return_code": 255
  }
}"
  traceback: "  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2actions/container/base.py", line 90, in _do_run
    runner.pre_run()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/paramiko_ssh_runner.py", line 145, in pre_run
    self._parallel_ssh_client = ParallelSSHClient(**client_kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/parallel_ssh.py", line 61, in __init__
    connect_results = self.connect(raise_on_any_error=raise_on_any_error)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/runners/parallel_ssh.py", line 91, in connect
    raise NoHostsConnectedToException(msg)
"
```

Running `ls -l ~stanley/.ssh/stanley_rsa` succeeds:
```
[root@st2vagrant ~]# ls -l ~stanley/.ssh/stanley_rsa
-rw-------. 1 stanley stanley 1679 Mar  1 17:44 /home/stanley/.ssh/stanley_rsa
[root@st2vagrant ~]# 
```

But of course this is because the shell is interpreting `~`, but ST2 can't.

I'm not sure what the best way to proceed here is. Couple of options I can think of:

1. Go back to using `/home/stanley` regardless of where stanley's $HOME dir is. Should still work to create that directory & set permissions. *BUT* make the authorized_keys line look like this:
`'cat /home/stanley/.ssh/stanley_rsa.pub >> ~stanley/.ssh/authorized_keys'` - the key there is that the private key can live pretty much anywhere as long as it has the right permissions. But when we put the pub key part in authorized_keys, we *must* put it in `~stanley/.ssh/authorized_keys`, not just `/home/stanley/.ssh/authorized_keys`.

2. Automatically determine the location of `~stanley`, translate it to the actual directory used, and then use `crudini` to set that option in `/etc/st2/st2.conf`.

Right now I'm leaning more towards option 1. Thoughts?